tests/macros.rs

```diff
@@ -0,0 +1,45 @@
+extern crate tera;
+
+use tera::{Context, Result, Tera};
+
+fn render_tpl(tpl_name: &str) -> Result<String> {
+    let tera = Tera::new("tests/macros/**/*.html").unwrap();
+
+    tera.render(tpl_name, &Context::new())
+}
+
+#[test]
+fn test_extend_and_include1() {
+    // trivial case: inheritance, only one macro import
+    let result = render_tpl("extend_and_include1.html");
+
+    assert_eq!(result.is_err(), false);
+    assert_eq!(result.unwrap(), "hello");
+}
+
+#[test]
+fn test_extend_and_include2() {
+    // inheritance, two macro imports, a and b, a is used, a uses self
+    let result = render_tpl("extend_and_include2.html");
+
+    assert_eq!(result.is_err(), false);
+    assert_eq!(result.unwrap(), "hello");
+}
+
+#[test]
+fn test_extend_and_include3() {
+    // inheritance, two macro imports, b and a, a is used, a uses self
+    let result = render_tpl("extend_and_include3.html");
+
+    assert_eq!(result.is_err(), false);
+    assert_eq!(result.unwrap(), "hello");
+}
+
+#[test]
+fn test_extend_and_include4() {
+    // like test_extend_and_include3 but no inheritance for completeness
+    let result = render_tpl("extend_and_include4.html");
```

Comment:
The name ("extend_...") is a bit odd because the file doesn't actually extend anything, but it's related to the "extend from stuff and include macros" test cases, so I kept them all consistent.